packages/@aws-cdk/aws-ecs/lib/fargate-queue-worker-service.ts

```diff
@@ -0,0 +1,76 @@
+import cdk = require('@aws-cdk/cdk');
+import { FargateService } from './fargate/fargate-service';
+import { FargateTaskDefinition } from './fargate/fargate-task-definition';
+import { QueueWorkerServiceBase, QueueWorkerServiceBaseProps } from './queue-worker-service-base';
+
+/**
+ * Properties to define a Fargate queue worker service
+ */
+export interface FargateQueueWorkerServiceProps extends QueueWorkerServiceBaseProps {
+  /**
+   * The number of cpu units used by the task.
+   * Valid values, which determines your range of valid values for the memory parameter:
+   * 256 (.25 vCPU) - Available memory values: 0.5GB, 1GB, 2GB
+   * 512 (.5 vCPU) - Available memory values: 1GB, 2GB, 3GB, 4GB
+   * 1024 (1 vCPU) - Available memory values: 2GB, 3GB, 4GB, 5GB, 6GB, 7GB, 8GB
+   * 2048 (2 vCPU) - Available memory values: Between 4GB and 16GB in 1GB increments
+   * 4096 (4 vCPU) - Available memory values: Between 8GB and 30GB in 1GB increments
+   *
+   * This default is set in the underlying FargateTaskDefinition construct.
+   *
+   * @default 256
+   */
+  readonly cpu?: string;
+
+  /**
+   * The amount (in MiB) of memory used by the task.
+   *
+   * This field is required and you must use one of the following values, which determines your range of valid values
+   * for the cpu parameter:
+   *
+   * 0.5GB, 1GB, 2GB - Available cpu values: 256 (.25 vCPU)
+   *
+   * 1GB, 2GB, 3GB, 4GB - Available cpu values: 512 (.5 vCPU)
+   *
+   * 2GB, 3GB, 4GB, 5GB, 6GB, 7GB, 8GB - Available cpu values: 1024 (1 vCPU)
+   *
+   * Between 4GB and 16GB in 1GB increments - Available cpu values: 2048 (2 vCPU)
+   *
+   * Between 8GB and 30GB in 1GB increments - Available cpu values: 4096 (4 vCPU)
+   *
+   * This default is set in the underlying FargateTaskDefinition construct.
+   *
+   * @default 512
+   */
+  readonly memoryMiB?: string;
```

Comment:
Not necessarily something to address here, but noting for posterity that we repeat these props (and their associated docstrings) a lot in the L3 constructs (both EC2 and Fargate) -- might be worth considering refactoring at some point, pending some of the discussion on how to handle these higher level constructs.

Reply:
Agreed, the only think we would need to consider is if the launchtype is Fargate, then it should default to (256 cpu or 512 memoryMiB) where as memory is be a required property for EC2 services (memoryLimit or memoryReservationLimit). It's not difficult, but we would be impacting readability and maintainability of the base classes. Thoughts?